### DISCUSSION TITLE
Listing additional pages within book chapters?

### DISCUSSION THREAD

**Discussion by mikemahoney218** [quarto-dev/quarto-cli#4827] [Feature Requests]
Hi all!

I'm trying to have a document listing inside a chapter of a book project. It works fine when the listing points at chapters of the book (defined in _quarto.yml), but doesn't work if they aren't explicitly included in the book structure. 

I don't want these items to appear in the table of contents, just in the listings section of this chapter. Is there a way to do that?

For instance, given a file structure:

```
chapter_1.qmd
posts/chapter_2.qmd
posts/not_a_chapter.qmd
```

Where `chapter_1` has the YAML:
```yaml
---
listing:
  contents: posts
---
```

And `_quarto.yml` includes:

```yaml
book:
  chapters:
    - chapter_1.qmd
    - posts/chapter_2.qmd
```

The listing in `chapter_1` will include chapter 2, but not `not_a_chapter`. 

This is a simplified example; in my actual case, there will be many items in the listing directory, and having them included with the book automatically (and not included in the chapter index) is my primary goal here (so "add them to the chapter structure" is rather self-defeating).

**Comment by dragonstyle** [comment]
You can do this, though listings will only render the listing for HTML output. I've attached an example project which shows this - the key is adding a `render` option to your project key which includes any files which are not in the book structure but which belongs in the project (this should be a list of globs).

In my example, something like:

```yaml
project:
  type: book
  render: 
    - resources

book:
  title: "Book"
  author: "Jane Doe"
  date: "3/14/2023"
  chapters:
    - index.qmd
    - intro.qmd
    - summary.qmd
    - references.qmd

bibliography: references.bib

format:
  html:
    theme: cosmo
  pdf:
    documentclass: scrreprt
```

and 

```markdown
---
listing:
  contents: resources/*.qmd
  type: table
---

# Summary

In summary, this book has no content whatsoever.

```


Sample Project: [book.zip](https://github.com/quarto-dev/quarto-cli/files/10972169/book.zip)

HTH!

**Reply by mikemahoney218** [reply]
Thanks @dragonstyle ! I don't think this quite works -- looking at your example (and my own situation), it looks like this generates a listing of the qmd files, not their rendered HTML:

![image](https://user-images.githubusercontent.com/38229299/225098238-e68415c0-70f2-45bd-ac8d-b5fd4dfd3d63.png)

Clicking the filename leads to downloading the qmd, not the HTML as you'd expect. Changing the glob to *.html produces an empty listing.

**Reply by dragonstyle** [reply]
🤣 That is kind of funny I completely overlooked that obvious and undesirable thing!!!

I tried a couple more variations on the theme and couldn't make listings really behave properly for this scenario in the context of a book - I think the issue boils down to the fact that books are generally designed to have constrained input (the chapters and appendices) since we do a lot of orchestration around those inputs to create the book and its behavior. They're really targeted at websites that can more easily absorb inputs that don't appear in the structure of the website directly.

Sorry I don't have a better idea!

**Reply by mikemahoney218** [reply]
Changed the category to `Idea` in the hopes it might be supported in the future :smile: This would be great for us; we've got an online "lab manual" book project which has chapters for long-running projects, and it'd be great to have a way to integrate short-lived updates into the book in the form of listings. We could probably get most of the way there with the website format instead, but the book format was pretty well adapted for our needs (minus the listings) right out of the box!
